scripts/get_graphids.py

```diff
@@ -0,0 +1,160 @@
+#! /usr/env python3
+
+import argparse
+import aiohttp
+import asyncio
+import async_timeout
+import posixpath
+from taskcluster.async import Index
+from taskcluster.async import Queue
+
+
+HG_URL = 'https://hg.mozilla.org/'
+
+REPOS = ('releases/mozilla-beta', 'releases/mozilla-release',
+         'releases/mozilla-esr52', 'mozilla-central')
+
+""" http://mozilla-version-control-tools.readthedocs.io/en/latest/hgmo/pushlog.html?highlight=json-pushes#json-pushes-command """  # noqa
+JSON_PUSHES = 'json-pushes?changeset={rev}'
+
+ACTION_INDEX = 'gecko.v2.{repo}.pushlog-id.{pushid}.actions'
+
+
+async def get_changeset_data(revision, session):
+    for repo in REPOS:
```

Comment:
If this stays a manual script, this is probably fine.
This would be more useful as a library function if it took a repo as an arg. We know what repo we're trying to promote against, so we don't have to cycle through each one.

Also, we're currently making these calls synchronously: make an async call, wait for it to finish, make the next async call. If the loop were outside the coroutine, we could queue up 4+ futures and await them all simultaneously, which results in concurrent requests:

```
tasks = []
for repo in REPOS:
    tasks.append(
        asyncio.ensure_future(
            get_changeset_data(repo, revision, session)
        )
    )
results = await raise_future_exceptions(tasks)
```

where `raise_future_exceptions` looks like [this](https://github.com/mozilla-releng/scriptworker/blob/master/scriptworker/utils.py#L300).

Reply:
I like this idea on paper, but I think the defining of raise_future_exceptions is bloating the utility script a bit more than I'd like for readability. The changesets are only a once-fetch attempt to pushlog so I'm not as worried about it. (per repo)